src/ExtUvLoop.php

```diff
@@ -313,4 +313,31 @@ private function createStreamListener()
 
         return $callback;
     }
+
+    /**
+     * @param float $interval
+     * @return int
+     */
+    private function convertFloatSecondsToMilliseconds($interval)
+    {
+        if ($interval < 0) {
+            throw new \InvalidArgumentException('Interval cannot be lower than 0.');
```

Comment:
This looks like a BC break to me and I'm surprised we don't seem to have a test case for this. What do you think about a solution like https://github.com/reactphp/event-loop/blob/babf91e65eea396681a8abda8ba91589a21da63b/src/StreamSelectLoop.php#L193-L194 with a test case for all loops instead?

Reply:
You are right, it is BC break. What about creating new ValueObject named `Interval` which will encapsulate all logic related to interval calculations?
Also IMO, passing interval lower than 0 should show deprecation notice, then in next major version there can be exception, WDYT?

Reply:
I agree that a value object might make sense in a higher level application, but keep in mind that this low level event loop is very performance sensitive, so I'd rather KISS. I also agree that a negative timeout doesn't make much "sense", but I'd rather leave the existing behavior and just make sure this means triggering the handler "asap".

It might make sense to reject this in the future, but I'm not sure it's worth a (subtle) BC break any time soon, so I'd rather continue this discussion in a separate ticket if you feel it's worth it :+1:

Reply:
I added `\React\Tests\EventLoop\AbstractLoopTest::testTimerIntervalBelowZeroRunsImmediately` - WDYT?

Reply:
Thank you, LGTM :+1: